lib/generator.ml

```diff
@@ -35,6 +42,21 @@ let split_elements file =
   | Some _ | None ->
     Error (`Invalid_atd_file file)
 
+let split_commit str =
+  match split_on_last_char ~char:':' str with
+  | Some (version, commit) ->
+    (match int_of_string_opt version with
+    | None ->
+      Error (`Invalid_version str)
+    | Some version ->
+      Ok (version, Some commit))
+  | None ->
+    (match int_of_string_opt str with
+    | None ->
+      Error (`Invalid_version str)
+    | Some version ->
+      Ok (version, None))
+
```

Comment:
we will make stork compatible with `version` as an int or as a string, so we will probably not check anymore if it can be converted to a string, but it's ok for now!

Reply:
It's a little wrong right now with what I wrote anyway I just noticed, because the error message refers to file names and I trigger the error for version/commit pairs.

Reply:
it's ok we'll fix that later!